packages/block-editor/src/components/inserter/block-patterns.js

```diff
@@ -37,11 +41,15 @@ function BlockPattern( { pattern, onClick } ) {
 			} }
 			tabIndex={ 0 }
 			aria-label={ pattern.title }
+			aria-describedby={ descriptionId }
 		>
 			<BlockPreview blocks={ blocks } viewportWidth={ viewportWidth } />
 			<div className="block-editor-inserter__patterns-item-title">
 				{ pattern.title }
 			</div>
+			<VisuallyHidden id={ descriptionId }>
```

Comment:
you can add 

```
{ !! pattern.description && (
// Your visibily hidden element
} )
```

to render this conditionally depending on the presence of the description.